### PR TITLE
add isDefined

### PR DESCRIPTION
**Pull request by andreineculau**
* Fixes:
* Breaking change: [ ]

---

implement https://github.com/lodash/lodash/issues/137

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -85,6 +85,16 @@ declare module 'lodash' {
...
 
+    /**
+     * Part of `lodash-firecloud`.
+     *
+     * Checks if value is defined.
+     *
+     * @param value The value to check.
+     * @return Returns true if value is defined, else false.
+     */
+    isDefined(value: T | undefined): value is T;
```
Comment (IanSavchenko) [inline src/index.d.ts:96]: What does `value is T` mean? I would expect `Boolean` here.

**Review by IanSavchenko — COMMENTED** [review]
Are you sure we need this? :)

File: src/index.d.ts (hunk 1 above)
Comment (andreineculau) [reply src/index.d.ts:96]: as per my (**limited**) understanding:

* T is a generic
* you return `true` (not just Boolean) if `value is T` and thus not `value is undefined`

original `isUndefined` for instance is defined similarly https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/lodash/common/lang.d.ts#L1504

similar pattern can be found here https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/lodash/fp.d.ts#L1908

**Comment by andreineculau** [comment]
> Are you sure we need this? :)

I count 15 occurrences of `!_.isUndefined` in our OSS repos, and they are all error-prone due to 1. `!_.` and 2. the apparent double negation (it is not, since undefined is a "type", but it reads as "if not not defined")

File: src/index.d.ts (hunk 1 above)
Comment (IanSavchenko) [reply src/index.d.ts:96]: ok, I see now

**Review by IanSavchenko — APPROVED** [review]
Ok

**Pull request merged**